app/components/channel_loader/index.js

```diff
@@ -1,32 +1,18 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
 
-import {markChannelAsRead, markChannelAsViewed} from 'mattermost-redux/actions/channels';
+import {getPostIdsInCurrentChannel} from 'mattermost-redux/selectors/entities/posts';
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
 
-import {handleSelectChannel, setChannelLoading} from 'app/actions/views/channel';
-
 import ChannelLoader from './channel_loader';
 
 function mapStateToProps(state, ownProps) {
     return {
-        channelIsLoading: ownProps.channelIsLoading || state.views.channel.loading,
+        channelIsLoading: ownProps.channelIsLoading || (state.views.channel.loading && !getPostIdsInCurrentChannel(state).length),
```

Comment:
Nothing really to do here right now, but having this come from either the parent or basing it off the store state seems confusing.